src/components/Flyout/Flyout.tsx

```diff
@@ -0,0 +1,165 @@
+
+import { FieldsetHeader } from "@compositions/FieldsetHeader/FieldsetHeader";
+import { Badge, BadgeProps } from "@elements/Badge/Badge";
+import { Button, ButtonStyle } from "@elements/Button/Button";
+import { Divider, DividerHeight } from "@elements/Divider/Divider";
+import IconCheck from "@elements/Icon/Generated/IconCheck";
+import { useButton } from "@react-aria/button";
+import { useDialog } from "@react-aria/dialog";
+import { FocusScope, useFocusRing } from "@react-aria/focus";
+import {
+    DismissButton,
+    OverlayContainer,
+    OverlayProvider,
+    useModal,
+    useOverlay,
+    useOverlayPosition,
+    useOverlayTrigger,
+} from "@react-aria/overlays";
+import { mergeProps } from "@react-aria/utils";
+import { useOverlayTriggerState } from "@react-stately/overlays";
+import { FOCUS_STYLE } from "@utilities/focusStyle";
+import { merge } from "@utilities/merge";
+import React, { Children, FC, forwardRef, HTMLAttributes, ReactNode, RefObject, useRef } from "react";
+import { useEffect } from "react";
+import { PropsWithChildren } from "react";
+import { watchModals } from "@react-aria/aria-modal-polyfill";
+
+export const FLYOUT_DIVIDER_COLOR = "#eaebeb";
+
+export type FlyoutProps = PropsWithChildren<{
+    trigger: ReactNode;
+    onClose: () => void;
+    onClick?: () => void;
+    title?: string;
+    decorator?: ReactNode;
+    badges?: BadgeProps[];
+}>;
+
+type OverlayProps = Omit<FlyoutProps, "trigger"> & {
+    isOpen?: boolean;
+    positionProps: HTMLAttributes<Element>;
+    overlayTriggerProps: HTMLAttributes<Element>;
+};
+const Overlay = forwardRef<HTMLDivElement, OverlayProps>(
+    ({ title, decorator, badges, onClick, onClose, children, isOpen, positionProps, overlayTriggerProps }, ref) => {
+        const { overlayProps } = useOverlay({ onClose, isOpen, isDismissable: true }, ref as RefObject<HTMLDivElement>);
+        const { modalProps } = useModal();
+        const { dialogProps, titleProps } = useDialog({}, ref as RefObject<HTMLDivElement>);
+
+        return (
+            <div
+                {...mergeProps(overlayProps, dialogProps, modalProps, positionProps, overlayTriggerProps)}
+                ref={ref}
+                className="tw-flex tw-flex-col tw-rounded tw-bg-white tw-text-black dark:tw-text-white dark:tw-bg-black-95 tw-shadow-mid tw-outline-none"
+            >
+                {title && (
+                    <>
+                        <div className="tw-flex tw-justify-between tw-flex-wrap tw-gap-3 tw-p-8">
+                            <div {...titleProps} className="tw-inline-flex">
+                                <FieldsetHeader decorator={decorator}>{title}</FieldsetHeader>
+                            </div>
+                            <div className="tw-inline-flex tw-gap-2 tw-flex-wrap">
+                                {badges &&
+                                    badges.map((badge, index) => <Badge key={`flyout-badge-${index}`} {...badge} />)}
```

Comment:
rename badge to badgeProps

```suggestion
                                    badges.map((badgeProps, index) => <Badge key={`flyout-badge-${index}`} {...badgeProps} />)}
```